### PR TITLE
add policy feature doc; improve protocol detection doc

### PR DESCRIPTION
**Pull request by wmorgan**
Signed-off-by: William Morgan <william@buoyant.io>

### CODE REVIEW

File: linkerd.io/content/2.11/features/server-policy.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+Linkerd's policy features allow you to *deny* traffic under certain conditions.
+It is configured with two basic mechanisms:
+
+1. A set of basic _default policies_, which can be set at the cluster,
+   namespace, workload, and pod level through Kubernetes annotations.
+2. `Server` and `ServerAuthorization` CRDs that specify fine-grained policy
+   for specific ports.
+
+These mechanisms work in conjunction. For example, a default cluster-wide
+policy of `deny-all` would prohibit any traffic to any meshed pod; traffic must
```
Comment (olix0r) [inline linkerd.io/content/2.11/features/server-policy.md:35]:
```suggestion
policy of `deny` would prohibit any traffic to any meshed pod; traffic must
```

File: linkerd.io/content/2.11/features/server-policy.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+### Policy annotations
+
+The `config.linkerd.io/default-inbound-policy` annotation can be set at a
+namespace, workload, and pod level, and will determine the default traffic
+policy at that point in the hierarchy. Valid default policies include:
+
+- `all-unauthenticated`: inbound proxies allow all connections
+- `all-authenticated`: inbound proxies allow only mTLS connections from other
+  meshed pods.
```
Comment (olix0r) [inline linkerd.io/content/2.11/features/server-policy.md:47]:
```suggestion
- `all-unauthenticated`: inbound proxies allow all connections
- `all-authenticated`: inbound proxies allow only mTLS connections from other
  meshed pods.
- `cluster-unauthenticated`: inbound proxies allow all connections from client IPs in the cluster's `clusterNetworks` (must be configured at install-time).
- `cluster-authenticated`: inbound proxies allow only mTLS connections from other
  meshed pods from IPs in the cluster's `clusterNetworks`.
```

File: linkerd.io/content/2.11/features/server-policy.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+initialization time, and thus, after a pod is created, changing the annotation
+will not change the default policy for that pod.
+
+### Policy CRDs
+
+The `Server` and `ServerAuthorization` CRDs further configure Linkerd's policy
+beyond the default policies. In contrast to annotations, these CRDs can be
+changed dynamically and policy behavior will be updated on the fly.
+
+A `Server` defines a port and a set of pods that is subject to policy. This set
```
Comment (olix0r) [inline linkerd.io/content/2.11/features/server-policy.md:67]:
```suggestion
A `Server` selects a port and a set of pods that is subject to policy. This set
```

File: linkerd.io/content/2.11/features/server-policy.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+of pods can correspond to a single workload, or to multiple workloads (e.g.
+port 4191 for every pod in a namespace). Once created, a `Server` resource
+denies all traffic to that port, and traffic to that port can only be enabled
+by creating `ServerAuthorization` CRDs.
```
Comment (olix0r) [inline linkerd.io/content/2.11/features/server-policy.md:71]:
```suggestion
by creating `ServerAuthorization` resources.
```

"creating a CRD" means we're creating the resource _definition_. We just want to create a resource _instance_.

**Review by olix0r — APPROVED** [review]
Not a blocker, but we probably want to describe how a `Server`'s `proxyProtocol` can be used to bypass (inbound) protocol detection.

**Review by mateiidavid — APPROVED** [review]
Looks good.

**Pull request merged**